src/coreclr/jit/llvm.cpp

```diff
@@ -447,7 +638,8 @@ Value* buildCnsInt(llvm::IRBuilder<>& builder, GenTree* node)
 
 Value* buildInd(llvm::IRBuilder<>& builder, GenTree* node, Value* ptr)
 {
-    return mapTreeIdValue(node->gtTreeID, builder.CreateLoad(castIfNecessary(builder, ptr, getLLVMTypeForVarType(node->TypeGet())->getPointerTo())));
+    // pass true to widen small ints to i32, TODO: will need to do the same for CLS_VAR, LCL_FLD
```

Comment:
> pass true to widen small ints to i32

An alternative design would be to pass `genActualType(node->TypeGet())` (for now, can be simplified to just `genActualType(node)` once you merge `main`) and not modify `getLLVMTypeForVarType`. `genActualType` maps all integers to `int/long` (see also `gtlist.h`).

Reply:
Thanks, that's better.